pkg/metrics/otel.go

```diff
@@ -29,324 +33,209 @@ const (
-      - include: remediate_duration_seconds
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [status]
-            aggregation_type: max
-      - include: kustomize_field_count
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [field_name]
-            aggregation_type: max
-      - include: kustomize_deprecating_field_count
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [deprecating_field]
-            aggregation_type: max
-      - include: kustomize_simplification_adoption_count
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [simplification_field]
-            aggregation_type: max
-      - include: kustomize_builtin_transformers
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [k8s_metadata_transformer]
-            aggregation_type: max
-      - include: kustomize_helm_inflator_count
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [helm_inflator]
-            aggregation_type: max
-      - include: kustomize_base_count
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [base_source]
-            aggregation_type: max
-      - include: kustomize_patch_count
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [patch_field]
-            aggregation_type: max
-      - include: kustomize_ordered_top_tier_metrics
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: [top_tier_field]
-            aggregation_type: max
-      - include: kustomize_resource_count
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: []
-            aggregation_type: max
-      - include: kustomize_build_latency
-        action: update
-        operations:
-          - action: aggregate_labels
-            label_set: []
-            aggregation_type: max
-extensions:
-  health_check:
-    endpoint: 0.0.0.0:13133
-service:
-  extensions: [health_check]
-  pipelines:
-    metrics/cloudmonitoring:
-      receivers: [opencensus]
-      processors: [batch, filter/cloudmonitoring, metricstransform/cloudmonitoring, resourcedetection]
-      exporters: [googlecloud]
-    metrics/prometheus:
-      receivers: [opencensus]
-      processors: [batch]
-      exporters: [prometheus]
-    metrics/kubernetes:
-      receivers: [opencensus]
-      processors: [batch, filter/kubernetes, metricstransform/kubernetes, resourcedetection]
-      exporters: [googlecloud/kubernetes]`
 )
+
+// CollectorConfigGooglecloudYAML returns the otel-collector-googlecloud config as YAML.
+func CollectorConfigGooglecloudYAML() (string, error) {
```

Comment:
It looks like valid configs can be generated programmatically, e.g. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/factory.go#L20

Would it be possible for us to use that to generate canonical/valid configs and then marshal to YAML?

Reply:
Tried it, and following this pattern will lead to adding many more dependencies, which could make the build larger and slower. It would also tightly couple us with the upstream design, complicating maintenance, especially since we have numerous exporters and processors. Given that the Config Sync ConfigMap is slightly customized and this change is a refactor, I prefer to keep it minimal while ensuring it remains relatively readable for future maintenance.

Reply:
Ack, sgtm